File: Ser2023/Company/Company.Web/Modules/ServerTypes/Store/OperationTypeRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface OperationTypeRow {
    OperationTypeId?: number;
    OpCode?: number;
    Operation?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class OperationTypeRow {
    static readonly idProperty = 'OperationTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'Operation';
    static readonly localTextPrefix = 'Store.OperationType';
    static readonly lookupKey = 'Store.OperationType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<OperationTypeRow>('Store.OperationType') }
    static async getLookupAsync() { return getLookupAsync<OperationTypeRow>('Store.OperationType') }

    static readonly deletePermission = 'Store:OperationType:Delete';
    static readonly insertPermission = 'Store:OperationType:Modify';
    static readonly readPermission = 'Store:OperationType:View';
    static readonly updatePermission = 'Store:OperationType:Modify';

    static readonly Fields = fieldsProxy<OperationTypeRow>();
}

File: Ser2023/Company/Company.Web/Modules/Store/OperationType/OperationTypeRow.cs

using Company.Administration.Entities;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace Company.Store;

[ConnectionKey("Store"), Module("Store"), TableName("[dbo].[OperationType]")]
[DisplayName("Operation Type"), InstanceName("Operation Type")]
[ReadPermission(PermissionKeys.OperationType.View)]
[ModifyPermission(PermissionKeys.OperationType.Modify)]
[DeletePermission(PermissionKeys.OperationType.Delete)]
[LookupScript("Store.OperationType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class OperationTypeRow : LoggingRow<OperationTypeRow.RowFields>, IIdRow, INameRow,  IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Operation Type Id"), Column("OperationTypeID"), Identity, IdProperty]
    public int? OperationTypeId
    {
        get => fields.OperationTypeId[this];
        set => fields.OperationTypeId[this] = value;
    }

    [DisplayName("Op Code"), NotNull]
    public short? OpCode
    {
        get => fields.OpCode[this];
        set => fields.OpCode[this] = value;
    }

    [DisplayName("Operation"), Size(30), NotNull, QuickSearch, NameProperty]
    public string Operation
    {
        get => fields.Operation[this];
        set => fields.Operation[this] = value;
    }


    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId
    {
        get => Fields.TenantId[this];
        set => Fields.TenantId[this] = value;
    }

    public Int32Field TenantIdField
    {
        get => Fields.TenantId;
    }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive
    {
        get => Fields.IsActive[this];
        set => Fields.IsActive[this] = value;
    }

    Int16Field IIsActiveRow.IsActiveField
    {
        get => Fields.IsActive;
    }

    #endregion Tenant & Activ


    public OperationTypeRow()
        : base()
    {
    }

    public OperationTypeRow(RowFields fields)
        : base(fields)
    {
    }

    public class RowFields : LoggingRowFields
    {
        public Int32Field OperationTypeId;
        public Int16Field OpCode;
        public StringField Operation;

        public Int16Field IsActive;
        public Int32Field TenantId;
    }
}
